Cesium3DTiles/src/RasterOverlayTileProvider.cpp

```diff
@@ -537,61 +590,11 @@ void RasterOverlayTileProvider::doLoad(
 
   this->beginTileLoad(tile, isThrottledLoad);
 
-  struct LoadResult {
-    RasterOverlayTile::LoadState state = RasterOverlayTile::LoadState::Unloaded;
-    CesiumGltf::ImageCesium image = {};
-    std::vector<Credit> credits = {};
-    void* pRendererResources = nullptr;
-  };
-
   this->loadTileImage(tile.getID())
-      .thenInWorkerThread(
-          [tileRectangle =
-               this->getTilingScheme().tileToRectangle(tile.getID()),
-           projection = this->getProjection(),
-           pPrepareRendererResources = this->getPrepareRendererResources(),
-           pLogger =
-               this->getLogger()](LoadedRasterOverlayImage&& loadedImage) {
-            if (!loadedImage.image.has_value()) {
-              SPDLOG_LOGGER_ERROR(
-                  pLogger,
-                  "Failed to load image:\n- {}",
-                  CesiumUtility::joinToString(loadedImage.errors, "\n- "));
-              LoadResult result;
-              result.state = RasterOverlayTile::LoadState::Failed;
-              return result;
-            }
-
-            if (!loadedImage.warnings.empty()) {
-              SPDLOG_LOGGER_WARN(
-                  pLogger,
-                  "Warnings while loading image:\n- {}",
-                  CesiumUtility::joinToString(loadedImage.warnings, "\n- "));
-            }
-
-            CesiumGltf::ImageCesium& image = loadedImage.image.value();
-
-            int32_t bytesPerPixel = image.channels * image.bytesPerChannel;
-
-            if (image.width > 0 && image.height > 0 &&
-                image.pixelData.size() >=
-                    static_cast<size_t>(
-                        image.width * image.height * bytesPerPixel)) {
-              void* pRendererResources =
-                  pPrepareRendererResources->prepareRasterInLoadThread(image);
-
-              LoadResult result;
-              result.state = RasterOverlayTile::LoadState::Loaded;
-              result.image = std::move(image);
-              result.credits = std::move(loadedImage.credits);
-              result.pRendererResources = pRendererResources;
-              return result;
-            }
-            LoadResult result;
-            result.pRendererResources = nullptr;
-            result.state = RasterOverlayTile::LoadState::Failed;
-            return result;
-          })
+      .thenInWorkerThread([this, tileId = tile.getID()](
+                              LoadedRasterOverlayImage&& loadedImage) {
+        return processLoadedImage(tileId, std::move(loadedImage));
```

Comment:
IMO this is an example of an "extract method" refactoring make the code less clear and harder to follow. When looking at `doLoad`, it's less clear what processing a loaded image actually entails, and in particular how it relates to the work that happens next in the main thread. And while looking at `processLoadedImage`, we're missing the vital context that it's invoked as a continuation in a worker thread. So, for example, that use of `this` may look fine in code review when actually it's pretty sus.